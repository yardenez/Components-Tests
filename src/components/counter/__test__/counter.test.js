import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer';
import Counter from './../counter';

afterEach(cleanup);

describe('Counter tests', () => {
    
    test("displays zero as the initial count value", ()=>{
        const {getByTestId}=render(<Counter/>);
        expect(getByTestId("count")).toHaveTextContent("0"); 
    });
    
    test("increments count when add button clicked",()=>{
        const {getByTestId}=render(<Counter/>);
        fireEvent.click(getByTestId("add-button"));
        expect(getByTestId("count")).toHaveTextContent("1"); 
    });
    
    test("decrements count when substract button clicked",()=>{
        const {getByTestId}=render(<Counter/>);
        fireEvent.click(getByTestId("substract-button"));
        expect(getByTestId("count")).toHaveTextContent("-1"); 
    });

    /* Will probably be used more on E2E tests, and not for each componenet  */
    test("Counter matches snapshot",()=>{
        const counter= renderer.create(<Counter/>);
        let tree = counter.toJSON();
        expect(tree).toMatchSnapshot();
    });
})



