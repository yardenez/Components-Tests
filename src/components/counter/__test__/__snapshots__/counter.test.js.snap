// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter tests Counter matches snapshot 1`] = `
<div>
  <span
    data-testid="count"
  >
    0
  </span>
  <div
    className="display-items-inline"
  >
    <button
      className="add-button"
      data-testid="add-button"
      onClick={[Function]}
    >
      + 
    </button>
    <button
      className="substract-button"
      data-testid="substract-button"
      onClick={[Function]}
    >
       - 
    </button>
  </div>
</div>
`;
